crates/subspace-networking/src/request_response_handler.rs

```diff
@@ -39,25 +42,63 @@ use sc_peerset::ReputationChange;
 use crate::request_responses::{IncomingRequest, OutgoingResponse, ProtocolConfig};
 const LOG_TARGET: &str = "request-response-handler";
 
+#[derive(Serialize, Deserialize, Debug)]
+pub struct Request {
+    pub start: PieceIndexHash,
+}
+
+impl From<Vec<u8>> for Request {
+    fn from(data: Vec<u8>) -> Request {
+        bincode::deserialize(&data).unwrap() // TODO
+    }
+}
```

Comment:
This is definitely a bad API, it should be `TryFrom`, but even then feels unnecessary, just provide a constructor. Also I'm not sure `start` field should be public in `Request` data structure.

Reply:
The comment refers to an early non-ready commit. Currently, the implementation is `TryFrom` as suggested. However, we discuss to change it to encode/decode methods.

Reply:
Fixed here: [74266e9](https://github.com/subspace/subspace/pull/541/commits/74266e94b9393b1ce550d1f0b9439233c1d24895)